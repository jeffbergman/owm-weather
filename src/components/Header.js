import React from 'react';
import { connect } from 'react-redux';
import Search from './Search';

const Header = (props) => {
  //this displays the title or city on the left
  //and a search component on the right
  return (
    <div className="header">
      <div className="header__left">
        {props.title}
      </div>
      <div className="header__right">
        <Search />  
      </div>
    </div>
  );
}

const mapStateToProps = (state) => {
  const title = state.weather.location ? state.weather.location.name : 'Weather';
  return {
    title
  }
};

export default connect(mapStateToProps)(Header);