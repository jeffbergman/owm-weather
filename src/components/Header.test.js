import React from 'react';
import { shallow } from 'enzyme';
import Header from './Header';

describe('Header', () => {

	let props = {
		title: 'New York'
	};

	let header = shallow(<Header.WrappedComponent {...props} />);

	it('renders correctly', () => {
		expect(header).toMatchSnapshot();
	});

	it('contains a connected Search component', () => {
		expect(header.find('Connect(Search)').exists()).toBe(true);
	});

	it('displays "New York" when props.title is "New York"', () => {
		expect(header.find('.header__left').text()).toEqual('New York');
	});
	
});


