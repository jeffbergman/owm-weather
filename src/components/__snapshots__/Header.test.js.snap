// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    title="New York"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="header__left"
>
          New York
</div>,
        <div
          className="header__right"
>
          <Connect(Search) />
</div>,
      ],
      "className": "header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "New York",
          "className": "header__left",
        },
        "ref": null,
        "rendered": "New York",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Connect(Search) />,
          "className": "header__right",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="header__left"
>
            New York
</div>,
          <div
            className="header__right"
>
            <Connect(Search) />
</div>,
        ],
        "className": "header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "New York",
            "className": "header__left",
          },
          "ref": null,
          "rendered": "New York",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Connect(Search) />,
            "className": "header__right",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
